posts/bonus-fastify-encapsulation.md

```diff
@@ -0,0 +1,320 @@
+# The Complete Guide to the Fastify Plugin System
+
+The most important feature of Fastify is the Plugin System and the encapsulation concept.
+These two features are synonymous as we are talking about the same thing.
+
+The Plugin System is a powerful tool that allows you to build a modular application.
+Unfortunately, it is also the most misunderstood feature of Fastify.
+This article will show how it works and how to avoid common mistakes.
+We are going to view the encapsulation with awesome images to make it easier to understand.
+
+
+## What is encapsulation?
+
+The encapsulation is not a new engineering concept nor a Fastify exclusive feature.
+It is the Plugin System base concept, and you can't ignore it if you want to take advantage of Fastify.
+
+You can read its formal description on the [Wikipedia page](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming).
+Briefly, it is a mechanism that allows you to create an isolated _context_ from the others application's _contexts_.
```

Comment:
_context_ from another application's _contexts_

the others is specific and I'm not sure what "the other" is, I think you meant another